### PR TITLE
Merged with develop

### PR DESCRIPTION
**Pull request by millerts**

### CODE REVIEW

File: item-migration/application.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ logging:
 ---
 spring:
   profiles: dev
-  datasource:
+  datasource:s
```
Comment (bdrainer) [inline item-migration/application.yml]: I bet the 's' was unintentional

Comment (millerts) [reply item-migration/application.yml]: fixed on your branch.  Nice catch.

File: common/src/main/java/org/opentestsystem/ap/common/datastore/DataStoreItemManager.java [hunk 2]
```diff
@@ -253,37 +254,16 @@ public ItemEntity saveMigratedItem(final ItemBankSystemUser systemUser,
         return persistedEntity;
     }
 
+    /**
+     * Deprecated - please use {@link ItemQueryService} instead
```
Comment (malaffoon) [inline common/src/main/java/org/opentestsystem/ap/common/datastore/DataStoreItemManager.java]: We should use `@deprecated` in the javadocs

File: common/src/main/java/org/opentestsystem/ap/common/datastore/DefaultItemQueryService.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+            throw new ResourceNotFoundException("Item " + itemId + " not found");
+        } else if (this.dataStoreUtility.isUserCreatingItem(user, masterEntity)) {
+            itemResponse = new ItemResponse(masterEntity.getItemJson(), ItemTransaction.builder()
+                .userName(masterEntity.getCreatedBy())
+                .commitId(masterEntity.getId())
+                .section(SECTION_CREATE)
+                .build());
+        } else if (masterEntity.isBeingEdited()) {
+            // someone else is creating the item
+            throw new ResourceNotFoundException("Item " + itemId + " not found");
```
Comment (malaffoon) [inline common/src/main/java/org/opentestsystem/ap/common/datastore/DefaultItemQueryService.java]: Should this be something else that results in a 409 (CONFLICT)? Looking at our handler that would be AcquireLockException.

File: common/src/main/java/org/opentestsystem/ap/common/datastore/DefaultItemQueryService.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    @Override
+    @Transactional(readOnly = true)
+    public List<ItemAssignment> findItemAssignments(String itemId) {
+        ItemEntity originatingEntity = this.dataStoreDataManager.findLastMaster(itemId);
+        Item item = this.dataStoreUtility.cloneItemJson(originatingEntity.getItemJson());
+        return new ArrayList<>(item.getAssignments());
+    }
+
+    @Override
+    @Transactional(readOnly = true)
```
Comment (malaffoon) [inline common/src/main/java/org/opentestsystem/ap/common/datastore/DefaultItemQueryService.java]: Why is this transactional, it makes a single call to the data manager?

File: common/src/main/java/org/opentestsystem/ap/common/datastore/DefaultItemQueryService.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+    @Override
+    public List<ItemTransaction> findItemTransactions(String itemId) {
+        List<ItemEntity> branches = this.dataStoreDataManager.findAllCurrentBranchEdits(itemId);
+        return CollectionUtils.isNotEmpty(branches)
+            ? branches.stream().map(mapToTransaction()).collect(toList())
+            : new ArrayList<>(INTEGER_ZERO);
+    }
+
+    @Override
+    @Transactional(readOnly = true)
```
Comment (malaffoon) [inline common/src/main/java/org/opentestsystem/ap/common/datastore/DefaultItemQueryService.java]: ditto (about transactional)

Hmm, maybe there is something i don't understand. Is the calling service required to know about implementation details in the data manager? Are transactional boundaries managed solely by the service layer? Seems ... brittle.

File: common/src/main/java/org/opentestsystem/ap/common/datastore/ItemQueryService.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+     * return that version of the item.  If the user is not editing the item then return the current master version of
+     * the item.  If the item is being created return NOT_FOUND if the requesting user is not the creator.
+     *
+     * @param user   The user making the request.
+     * @param itemId The item to find.
+     * @return The item and any open transactions associated with the item.
+     */
+    ItemResponse findItem(ItemBankUser user, String itemId);
+
+    List<ItemAssignment> findItemAssignments(String itemId);
```
Comment (malaffoon) [inline common/src/main/java/org/opentestsystem/ap/common/datastore/ItemQueryService.java]: Not sure if this is just cut&paste but there are a lot of javadocs missing. What is the order of the return? For the next method (findUserTransaction), does it return null or throw an exception if the transaction can't be found? et cetera

File: common/src/main/java/org/opentestsystem/ap/common/management/ItemManager.java [hunk 7]
```diff
@@ -222,7 +158,86 @@ public EducatorCommitteeReview saveEducatorCommitteeReviewFeedback(ItemBankUser
      * @param branch The branch
      */
     private void beginEditErrorIfNotAuthorized(ItemBankUser user, String itemId, String branch) {
-        Item item = this.dataStoreManager.getCurrentMaster(itemId);
+        Item item = itemQueryService.getCurrentMaster(itemId);
         PermissionValidator.validateCanEditItemSection(user, item, branch);
     }
+
+    /**
+     * Deprecated - please use {@link org.opentestsystem.ap.common.datastore.ItemQueryService}
```
Comment (malaffoon) [inline common/src/main/java/org/opentestsystem/ap/common/management/ItemManager.java]: ditto (use @deprecated)

**Comment by malaffoon** [comment]
If translation service needs rabbit, can you update docker-compose.yml to reflect that? That is, add a `depends_on` entry for rabbitmq in item-translation-service.

File: common/src/main/java/org/opentestsystem/ap/common/datastore/DefaultItemQueryService.java (hunk 3 above)
Comment (millerts) [reply common/src/main/java/org/opentestsystem/ap/common/datastore/DefaultItemQueryService.java]: I defer to @bdrainer.  I copied this code "as-is" from the existing manager class.

File: common/src/main/java/org/opentestsystem/ap/common/datastore/DefaultItemQueryService.java (hunk 5 above)
Comment (millerts) [reply common/src/main/java/org/opentestsystem/ap/common/datastore/DefaultItemQueryService.java]: I copied this code "as-is" from the manager.  I believe there are inconsistencies in the use of transactional in this area of the code.

File: common/src/main/java/org/opentestsystem/ap/common/datastore/ItemQueryService.java (hunk 6 above)
Comment (millerts) [reply common/src/main/java/org/opentestsystem/ap/common/datastore/ItemQueryService.java]: I copied this code "as-is" from the manager.  When there were JavaDocs I added them.  I did not add missing JavaDocs.

**Pull request merged**